sdk/nodejs/resource.ts

```diff
@@ -220,6 +220,27 @@ export abstract class Resource {
             throw new ResourceError("Missing resource name argument (for URN creation)", opts.parent);
         }
 
+        // Before anything else - if there are transformations registered, invoke them in order to transform the properties and
+        // options assigned to this resource.
+        const parent = opts.parent || getStackResource() || { __transformations: undefined };
+        this.__transformations = [ ...(opts.transformations || []), ...(parent.__transformations || []) ];
+        for (const transformation of (this.__transformations || [])) {
```

Comment:
```suggestion
        for (const transformation of this.__transformations) {
```

The above line ensures that it's not null.